syntax = "proto3";

service RouteClassify {
    rpc GetDomains(Empty) returns (Domains) {}
    rpc GetClassif(TextToClassify) returns (TextClassified) {}
    rpc RouteClassify(stream TextToClassify) returns (stream TextClassified) {}
}

message Empty {
}

message Domains {
    repeated string domains = 1;
}

message TextToClassify {
  string text = 1;
  string language = 2;
  int32 count = 3;
  float threshold = 4;
}

message TextClassified {
  string text = 1;
  string language = 2;
  int32 count = 3;
  float threshold = 4;
  repeated Score intention = 5;
}

message Score{
    string label = 1;
    float confidence = 2;
}